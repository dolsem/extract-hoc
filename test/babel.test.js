const path = require('path');
const fs = require('fs');
const { transformFileSync } = require('babel-core');

const FIXTURES_DIR = path.join(__dirname, '__fixtures__');

function trim(str) {
  return str.replace(/^\s+|\s+$/, '')
}

function specsFromFixtures(cb) {
  fs.readdirSync(FIXTURES_DIR).forEach(fixtureName => {
    const fixtureFile = path.join(FIXTURES_DIR, fixtureName)

    if (fs.statSync(fixtureFile).isFile()) {
      const caseName = fixtureName.replace(/\.js$/, '').split('-').join(' ')

      it(`for ${caseName}`, () => cb({ fixtureFile }))
    }
  })
}

describe('extract HoCs', () => specsFromFixtures(({ fixtureFile }) => {
  const actual = transformFileSync(fixtureFile).code
  const codeWithoutFilename = actual.replace(
    new RegExp(`["']${fixtureFile}["']`, 'g'),
    '__FILENAME__',
  );

  expect(trim(codeWithoutFilename)).toMatchSnapshot();
}));

describe('extract HoCs with copyExactName', () => specsFromFixtures(({ fixtureFile }) => {
  const actual = transformFileSync(fixtureFile, { plugins: [['../../lib/babel', { copyExactName: true }]] }).code
  const codeWithoutFilename = actual.replace(
    new RegExp(`["']${fixtureFile}["']`, 'g'),
    '__FILENAME__',
  );

  expect(trim(codeWithoutFilename)).toMatchSnapshot();
}));
