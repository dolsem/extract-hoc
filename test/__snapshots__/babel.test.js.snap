// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extract HoCs for export default declaration 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});

var _default_arg, _default_arg2;

var level0 = 1;
_default_arg = baz()(level0)
_default_arg2 = bar()(_default_arg)
exports.default = foo()(_default_arg2);"
`;

exports[`extract HoCs for export named declaration 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});

var _final_arg, _final_arg2;

var level0 = 1;
var _final_arg = baz()(level0),
    _final_arg2 = bar()(_final_arg),
    final = exports.final = foo()(_final_arg2);"
`;

exports[`extract HoCs for variable declarator 1`] = `
"\\"use strict\\";

var _final_arg, _final_arg2;

var level0 = 1;
var _final_arg = baz()(level0),
    _final_arg2 = bar()(_final_arg),
    final = foo()(_final_arg2);"
`;

exports[`extract HoCs with copyExactName for export default declaration 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});

var _default_arg;

var level0 = 1;
_default_arg = {
  default: bar()({
    default: baz()(level0)
  }.default)
}.default
exports.default = foo()(_default_arg);"
`;

exports[`extract HoCs with copyExactName for export named declaration 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});

var _final_arg;

var level0 = 1;
var _final_arg = {
  final: bar()({
    final: baz()(level0)
  }.final)
}.final,
    final = exports.final = foo()(_final_arg);"
`;

exports[`extract HoCs with copyExactName for variable declarator 1`] = `
"\\"use strict\\";

var _final_arg;

var level0 = 1;
var _final_arg = {
  final: bar()({
    final: baz()(level0)
  }.final)
}.final,
    final = foo()(_final_arg);"
`;
